contracts/Core/BlockManager.sol

```diff
@@ -173,27 +174,38 @@ contract BlockManager is Initializable, ACL, BlockStorage {
         }
     }
 
-    function confirmBlock() external initialized onlyRole(parameters.getBlockConfirmerHash()) {
+    function claimBlockReward() public initialized checkState(parameters.confirm()){
+        uint256 epoch = parameters.getEpoch();
+        uint256 stakerId = stakeManager.getStakerId(msg.sender);
+        require(stakerId > 0, "Structs.Staker does not exist");
+        require(blocks[epoch].proposerId == 0, "Block already confirmed");
+
+        for (uint8 i = 0; i < proposedBlocks[epoch].length; i++) {
```

Comment:
every proposed block is confirmed here?

Reply:
Nope, we come out of the for loop in L191. It works same as this https://github.com/razor-network/contracts/blob/034b303a4174ae846e0ed21b013b90c68bd41715/contracts/Core/BlockManager.sol#L181

Reply:
https://github.com/SamAg19/contracts/blob/f4223bc5fe142098309de004ddaf9f8a80f792da/contracts/Core/BlockManager.sol#L181 makes sure that this function is called only once and cant be called once a block is confirmed

Reply:
the if doesnt need to be that big. it can just be if (!proposedBlocks[epoch][i].valid) return;

same in https://github.com/razor-network/contracts/blob/034b303a4174ae846e0ed21b013b90c68bd41715/contracts/Core/BlockManager.sol#L181